### PR TITLE
[12] - Implement the authorization/authentication functionality

### LINKED ISSUE #12: Implement the Authorization/Authentication functionality
We need to introduce the authorization/authentication functionality on the project

- [x] Introduce the authorization/authentication structure on the project
- [x] Introduce the login/register/logout/refresh token functionalities

### PR DESCRIPTION
**Pull request by spatiszisis**
- Added some new libraries that needed for this issue. (e.g. spring-boot-starter-security etc.)
- Introduced some new routes for the new auth operations
- Introduced a whole new functionality for the Authorization/Authentication process of the project
- Introduced a new entity to store the tokens of the users.

### CODE REVIEW

**Review by EmmanSkout — APPROVED** [review]
In general, very well implemented and clean code. 

I like the use you are making of annotations and more specifically Lombok . 
Maybe we could cut down the use of "var" . In my opinion , java is meant to be strongly typed and we should stick to that. 

Other than that, I'm happy with this request and I'm approving.

**Review by apokiou — APPROVED** [review]
Looks Good Zisis!

File: src/main/java/net/trapezokomos/dashboard/controller/ConsumerController.java
Comment (christosavramis) [inline src/main/java/net/trapezokomos/dashboard/controller/ConsumerController.java:1]: Change request param annotation to path variable when applicable e.g
 @RequestParam(value = "id", required = true) Long id 
to @PathVariable Long id



request param is ?param
path variable param is /param/
also in general avoid setting default values e.g. required = true

Comment (spatiszisis) [reply src/main/java/net/trapezokomos/dashboard/controller/ConsumerController.java:1]: OK

**Pull request merged**